File: app/src/main/java/com/anthonymandra/rawdroid/data/entity/SubjectEntity.java
package com.anthonymandra.rawdroid.data.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.provider.BaseColumns;

@Entity(tableName = SubjectEntity.DATABASE)
public class SubjectEntity extends PathEntity
{
	public static final String DATABASE = "xmp_subject";
	public static final String _ID = BaseColumns._ID;
	public static final String NAME = "name";
	public static final String RECENT = "recent";

	@PrimaryKey(autoGenerate = true)
	@ColumnInfo(name = _ID)
	public long id;

	@ColumnInfo(name = NAME)
	public String name;

	@ColumnInfo(name = RECENT)
	public String recent;


}

File: app/src/main/java/com/anthonymandra/curator/data/SubjectEntity.kt
package com.anthonymandra.curator.data

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(
    tableName = "xmp_subject",
    indices = [ Index(value = [ "id" ]) ])
data class SubjectEntity(
    var name: String = "",
    var recent: Long = 0,
    // PathEntity
    @Ignore override var id: Long = 0,
    @Ignore override var path: String = "",
    @Ignore override var parent: Long = -1,
    @Ignore override var depth: Int = 0): PathEntity(), Parcelable
